core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java

```diff
@@ -699,7 +701,16 @@ else if (Tables.getTableState(context.getInstance(), table) == TableState.OFFLIN
 
     }
 
-    void addMutations(MutationSet mutationsToSend) {
+    void queueMutations(final MutationSet mutationsToSend) {
+      queueThreadPool.submit(new Runnable() {
```

Comment:
Ultimately, I really don't have that strong of an opinion here. I think using the ExecutorService as a queue for MutationSets is a smell (instead of directly maintaining such a queue), but I can understand the argument for using the ExecutorService *because* it encompasses both approaches. I don't think there's a big reason one way or the other; I simply think that the pseudo-code outline above is clearer than Dave's current changes. If Keith and Dave are happy, then merge it and move on.